### ISSUE TITLE
[BUG] error: expected a set but found a list

### ISSUE THREAD

**Issue by LudovicoPiero** [nix-community/nixvim#1702]
| Field        | Description |
|--------------|-------------|
| Plugin       | `lsp`  | 
| Nixpkgs      | `unstable` | 
| Home Manager | `unstable` | 

- [x] I have read the [FAQ](https://nix-community.github.io/nixvim/user-guide/faq.html) and my bug is not listed there.

## Description

Error after doing `nix flake update`. Here's the error message
```console
error:
       … while calling the 'head' builtin
         at /nix/store/m8869v1img6mqn1d52dyp75v4f3sf1qy-source/lib/attrsets.nix:1575:11:
         1574|         || pred here (elemAt values 1) (head values) then
         1575|           head values
             |           ^
         1576|         else

       … while evaluating the attribute 'value'
         at /nix/store/m8869v1img6mqn1d52dyp75v4f3sf1qy-source/lib/modules.nix:809:9:
          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          810|         inherit (res.defsFinal') highestPrio;

       … while evaluating the option `system.build.toplevel':

       … while evaluating definitions from `/nix/store/m8869v1img6mqn1d52dyp75v4f3sf1qy-source/nixos/modules/system/activation/top-level.nix':

       … while evaluating the option `warnings':

       … while evaluating definitions from `/nix/store/m8869v1img6mqn1d52dyp75v4f3sf1qy-source/nixos/modules/system/boot/systemd.nix':

       … while evaluating the option `systemd.services.home-manager-airi.serviceConfig':

       … while evaluating definitions from `/nix/store/cx36lz9m7vv8qznmvxkzg2bs4hlgpqgf-source/src/transformers/nixosConfigurations.nix':

       … while evaluating the option `home-manager.users.airi.home.file.".manpath".source':

       … while evaluating definitions from `/nix/store/6nfmqa5gw9gs1wbxfw9p0qpk5apblb7g-source/modules/files.nix':

       … while evaluating the option `home-manager.users.airi.home.file.".manpath".text':

       … while evaluating definitions from `/nix/store/6nfmqa5gw9gs1wbxfw9p0qpk5apblb7g-source/modules/programs/man.nix':

       … while evaluating the option `home-manager.users.airi.programs.nixvim.content':

       … while evaluating definitions from `/nix/store/fg9vjx7bl5yj0slbwgw1rjghmlvvi83f-source/modules/output.nix':

       … while evaluating the option `home-manager.users.airi.programs.nixvim.extraConfigLua':

       … while evaluating definitions from `/nix/store/fg9vjx7bl5yj0slbwgw1rjghmlvvi83f-source/plugins/lsp':

       … while evaluating the option `home-manager.users.airi.programs.nixvim.plugins.lsp.servers.rust-analyzer.settings':

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: expected a set but found a list: [ { properties = { "rust-analyzer.cargoRunner" = { type = [ "null" "string" ]; default = null; description = "Custom cargo runner extension ID."; }; "rust-analyzer.discoverProjectRunner" = { type = [ "null" "string" ]; default = null; markdownDescription = "Sets the extension responsible for determining which extension the rust-analyzer extension uses to generate `rust-project.json` files. This should should only be used\n if a build system like Buck or Bazel is also in use."; }; "rust-analyzer.restartServerOnConfigChange" = { type = "boolean"; default = false; «4294967289 attributes elided» }; «4294967292 attributes elided» }; «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } «154 items elided» ]
```

## Minimal, Reproducible Example (MRE)

```nix
programs.nixvim = {
    plugins = {
      lsp = {
        enable = true;
        servers = {
          rust-analyzer = {
            enable = true;
            filetypes = [ "rs" ];
            installCargo = true;
            installRustc = true;

            settings = {
              diagnostics = {
                enable = true;
                # experimental.enable = true;
                styleLints.enable = true;
              };

              files = {
                excludeDirs = [
                  ".direnv"
                  "rust/.direnv"
                ];
              };

              inlayHints = {
                bindingModeHints.enable = true;
                closureStyle = "rust_analyzer";
                closureReturnTypeHints.enable = "always";
                discriminantHints.enable = "always";
                expressionAdjustmentHints.enable = "always";
                implicitDrops.enable = true;
                lifetimeElisionHints.enable = "always";
                rangeExclusiveHints.enable = true;
              };

              procMacro = {
                enable = true;
              };
            };
          };
        };
      };
    };
 };
}
```

**Comment by MattSturgeon** [comment]
The error is that a list has been assigned to `plugins.lsp.servers.rust-analyzer.settings`, however that doesn't appear to be anywhere obvious in _your_ config 🤔 
 The actual value assigned looks like it's come from our update script:
```nix
[ { properties = { "rust-analyzer.cargoRunner" = { type = [ "null" "string" ]; default = null; description = "Custom cargo runner extension ID."; }; "rust-analyzer.discoverProjectRunner" = { type = [ "null" "string" ]; default = null; markdownDescription = "Sets the extension responsible for determining which extension the rust-analyzer extension uses to generate `rust-project.json` files. This should should only be used\n if a build system like Buck or Bazel is also in use."; }; "rust-analyzer.restartServerOnConfigChange" = { type = "boolean"; default = false; «4294967289 attributes elided» }; «4294967292 attributes elided» }; «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } { «4294967286 attributes elided» } «154 items elided» ]
```

Without looking into this properly, my first thought is that one of my recent refactoring PRs broke something here.

**Comment by LudovicoPiero** [comment]
Actually, it's only happening on `nixos-unstable-small`. On `nixos-unstable`, I get the bash-language-server error.

**Comment by MattSturgeon** [comment]
> On `nixos-unstable`, I get the bash-language-server error.

That's a separate issue (see #1699). Removing `inputs.nixvim.follows` lines will work-around that.

**Comment by khaneliman** [comment]
Getting this error too on my config all of a sudden.

**Comment by s1341** [comment]
I'm also getting this.. is there a fix or a workaround?

**Comment by ngn999** [comment]
> I'm also getting this.. is there a fix or a workaround?

Same error here.
Comment nixvim.inputs.nixpkgs.follows , then `nix flake lock` works for me.

**Comment by pta2002** [comment]
Getting this even with rust-analyzer disabled.

**Comment by GaetanLepage** [comment]
A quick fix will be shipped in #1719

**Comment by GaetanLepage** [comment]
Fixed (kind of) in #1719.

I will leave this issue open because we need to actually fix it for real.

**Comment by ZainKergaye** [comment]
Got the same error, thanks for giving out a solution.  @pta2002 Have you updated your system fully? I'm still new and don't know all the key terms but at first that happened but a nix flake update and rebuild put me at the latest nixvim release

**Comment by dezren39** [comment]
i no longer am experiencing this error

**Comment by MattSturgeon** [comment]
> I no longer am experiencing this error

As mentioned above https://github.com/nix-community/nixvim/issues/1702#issuecomment-2175698064, we have a temporary workaround in place. @traxys is looking into the actual issue, so hopefully it'll be fixed _properly_ soon.

**Comment by GaetanLepage** [comment]
This is now fully fixed right ?

**Comment by MattSturgeon** [comment]
Should be fixed by #1806
